### ISSUE TITLE
Questions and thoughts 

### ISSUE THREAD

**Issue by datnamer** [dblairchappell/Prigogine#1]
Hello,

I am a python user with an eye on Julia and was following the development of your Julia ABM package that I had stumbled upon.  

I have some questions and ideas:
- Any particular reason you switched to python? Either way is cool, just curious.
- Also there is another recent Python ABM toolkit that  looks promising: https://github.com/projectmesa/mesa 
  Perhaps it would be better to combine efforts and create one substantially awesome package?
- If you miss Julia's low cost types/abstractions, Numba, the python numeric JIT compiler with comparable speed to c, is soon getting JIT classes: https://groups.google.com/a/continuum.io/forum/#!topic/numba-users/8SMvZfhVJkc 
- Finally, [Dask](http://dask.pydata.org/en/latest/) provides threaded and multiprocess schedulers that can make numeric parallel and out of core computing much easier (Scipy talk [here](https://www.youtube.com/watch?v=1kkFZ4P-XHg)) Numba can release the gill so using that with dask you can probably do  robust shared memory parallel abm.

Just some thoughts. Thanks!

**Comment by dblairchappell** [comment]
Hi datnamer. Thank you for your questions. I only just saw them today so apologies for not responding sooner.

On the first of them, I switched from Julia for two reasons really. The first was that I wanted to take advantage of the ANTLR4 parser generator for developing a language syntax in a more convenient way. As ANTLR4 is not capable of targeting Julia, I was forced into switching to Python. The second reason was to do with Julia being a relatively new language and so subject to possible changes that could leave any work I do here deprecated.

On the second question, I hadn't heard of Mesa but it looks like an interesting project. As it stands though, I am developing Prigogine as part of a software development module at college, so it's likely to remain it's own thing for the time being.

In regards to Numba, it was one of the solutions I was looking at for making Prigogine as fast as possible. I might yet still take advantage of it, but for now I am using NumPy to store agent data and carry out vectorised array operations. It seems to be producing some good results from some initial tests I've carried out in the past day or two.

Finally, I also hadn't heard of Dask so I'm glad you bought it to my attention because I'm due to start parallelising the code in the next couple of months.

Once again, thanks for taking interest in what I'm doing; I hope I've managed to answer your questions. If you have any others, or just comments in general, then ask away.

All the best, Dave. :)
